tests/jballerina-unit-test/src/test/resources/test-src/expressions/invocations/function-invocation-expr.bal

```diff
@@ -354,6 +354,57 @@ function testTypeRefTypedRestArg() {
 function testFn(int a, string... b) returns int =>
     b.reduce(function (int tot, string str) returns int => tot + str.length(), a);
 
+function testFuncWithNilReturnTypeWithoutVariableAssignment() {
```

Comment:
Please add some additional test cases for scenarios where the return type is used as a type reference.

Reply:
what do we hope to test with this... can you provide an example

Reply:
I mean to add a test for [this](https://github.com/ballerina-platform/ballerina-lang/pull/39819#discussion_r1134903418) kind of scenario